### PR TITLE
[MM-8737] Add Go code samples to API reference for user endpoints

### PR DESCRIPTION
**Pull request by z4cco**
Solves ticket [MM-8737](https://mattermost.atlassian.net/browse/MM-8737)

Fixes https://github.com/mattermost/mattermost-server/issues/10606

### CODE REVIEW

File: v4/source/users.yaml [hunk 1]
```diff
@@ -888,6 +1144,17 @@
                 type: boolean
         '400':
           $ref: '#/responses/BadRequest'
+      x-code-samples:
+        - lang: 'Go'
+          source: |
+            import "github.com/mattermost/mattermost-server/model"
+
+            Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
```
Comment (hanzei) [inline v4/source/users.yaml]: This doesn't have to be authenticated.

File: v4/source/users.yaml [hunk 2]
```diff
@@ -962,6 +1242,17 @@
           $ref: '#/responses/Unauthorized'
         '403':
           $ref: '#/responses/Forbidden'
+      x-code-samples:
+        - lang: 'Go'
+          source: |
+            import "github.com/mattermost/mattermost-server/model"
+
+            Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
```
Comment (hanzei) [inline v4/source/users.yaml]: This doesn't have to be authenticated.

File: v4/source/users.yaml [hunk 3]
```diff
@@ -1175,6 +1533,17 @@
             $ref: '#/definitions/StatusOK'
         '400':
           $ref: '#/responses/BadRequest'
+      x-code-samples:
+        - lang: 'Go'
+          source: |
+            import "github.com/mattermost/mattermost-server/model"
+
+            Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
```
Comment (hanzei) [inline v4/source/users.yaml]: This doesn't have to be authenticated.

File: v4/source/users.yaml [hunk 4]
```diff
@@ -1204,6 +1573,17 @@
...
```
Comment (hanzei) [inline v4/source/users.yaml]: This doesn't have to be authenticated.

File: v4/source/users.yaml [hunk 5]
```diff
@@ -1272,6 +1652,30 @@
           $ref: '#/responses/NotFound'
         '501':
           $ref: '#/responses/NotImplemented'
+      x-code-samples:
+        - lang: 'Go'
+          source: |
+            import "github.com/mattermost/mattermost-server/model"
+
+            Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
```
Comment (hanzei) [inline v4/source/users.yaml]: This doesn't have to be authenticated.

File: v4/source/users.yaml [hunk 6]
```diff
@@ -346,6 +393,15 @@
...
           $ref: '#/responses/NotFound'
+      x-code-samples:
+        - lang: 'Go'
+          source: |
+            import "github.com/mattermost/mattermost-server/model"
+
+            Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
+
+            stats, resp := Client.GetTotalUsersStats("").TotalUsersCount
```
Comment (hanzei) [inline v4/source/users.yaml]: I would like to stay consistent here and just get the response.
```suggestion
            stats, resp := Client.GetTotalUsersStats("")
```

File: v4/source/users.yaml [hunk 7]
```diff
@@ -508,6 +619,30 @@
...
+            patch.Username = "testUsername"
+            patch.FirstName = "testFirstname"
+            patch.LastName = "testLastname"
+            patch.Nickname = "testNickname"
+            patch.Locale = "en"
+            patch.Position = "testPosition"
+            patch.Props = model.StringMap{}
+            patch.Props["testPropKey"] = "testPropValue"
+            patch.NotifyProps = model.StringMap{}
+            patch.NotifyProps = ["comment"] = "somethingrandom"
```
Comment (hanzei) [inline v4/source/users.yaml]:
```suggestion
            patch.NotifyProps["comment"] = "somethingrandom"
```

File: v4/source/users.yaml [hunk 8]
```diff
@@ -1272,6 +1652,30 @@
...
+
+            currentService := "email"
+            newService := "gitlab"
+            email := "test@domain.com"
+            password := "awesomePassword"
+            mfaCode := "adWv1qPZmHdtxk7Lmqh6RtxWxS"
+            ldapLoginID := "RdDjEDlkWgt7ndjyVLwWGvnX8c"
+
+
+            link, resp := Client.SwitchAccountType(&model.SwitchRequest{
```
Comment (hanzei) [inline v4/source/users.yaml]: This is inconsistent with how you did this before. Would you please fix this?

File: v4/source/users.yaml [hunk 9]
```diff
@@ -68,6 +68,7 @@
             import "github.com/mattermost/mattermost-server/model"
 
             Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
```
[the remark this answers is not in the document]
Comment (z4cco) [reply v4/source/users.yaml]: done

File: v4/source/users.yaml (hunk 1 above)
Comment (z4cco) [reply v4/source/users.yaml]: done

File: v4/source/users.yaml (hunk 2 above)
Comment (z4cco) [reply v4/source/users.yaml]: done

File: v4/source/users.yaml (hunk 3 above)
Comment (z4cco) [reply v4/source/users.yaml]: done

File: v4/source/users.yaml (hunk 4 above)
Comment (z4cco) [reply v4/source/users.yaml]: done

File: v4/source/users.yaml (hunk 5 above)
Comment (z4cco) [reply v4/source/users.yaml]: done

File: v4/source/users.yaml (hunk 6 above)
Comment (z4cco) [reply v4/source/users.yaml]: done

File: v4/source/users.yaml (hunk 7 above)
Comment (z4cco) [reply v4/source/users.yaml]: done

File: v4/source/users.yaml (hunk 8 above)
Comment (z4cco) [reply v4/source/users.yaml]: done

**Pull request merged**